### PR TITLE
 Update TypeArray Detachbuffer

### PR DESCRIPTION
**Pull request by kisbg**
related test262 test-cases has been removed from skiplist

Signed-off-by: bence gabor kis <kisbg@inf.u-szeged.hu>

### CODE REVIEW

**Review by ksh8281 — APPROVED** [review]

File: src/interpreter/ByteCodeInterpreter.cpp [hunk 1]
```diff
@@ -3972,8 +3973,10 @@ NEVER_INLINE void ByteCodeInterpreter::setObjectOpcodeSlowCase(ExecutionState& s
     if (willBeObject.isPrimitive()) {
         obj->preventExtensions(state);
     }
-
     bool result = obj->setIndexedProperty(state, property, registerFile[code->m_loadRegisterIndex]);
+    if (obj->isTypedArrayObject() && obj->asTypedArrayObject()->buffer()->isDetachedBuffer()) {
+        result = true;
+    }
```
Comment (clover2123) [inline src/interpreter/ByteCodeInterpreter.cpp:3979]: IMO this part can be removed because you already fixed `setIndexedProperty` of TypedArray according to the new standard.

File: src/runtime/GlobalObjectBuiltinTypedArray.cpp [hunk 2]
```diff
@@ -567,11 +567,14 @@ static Value builtinTypedArrayIndexOf(ExecutionState& state, Value thisValue, si
     }
     size_t k = (size_t)doubleK;
 
+    if (O->asTypedArrayObject()->buffer()->isDetachedBuffer()) {
+        return Value(-1);
+    }
```
Comment (clover2123) [inline src/runtime/GlobalObjectBuiltinTypedArray.cpp:572]: Why did you add this part?
I cannot find any matching point in the spec.

File: src/runtime/GlobalObjectBuiltinTypedArray.cpp [hunk 3]
```diff
@@ -1264,8 +1264,10 @@ static Value builtinTypedArrayJoin(ExecutionState& state, Value thisValue, size_
     ToStringRecursionPreventerItemAutoHolder holder(state, O);
 
     StringBuilder builder;
-    Value elem = O->getIndexedProperty(state, Value(0)).value(state, O);
-    RELEASE_ASSERT(!elem.isUndefinedOrNull());
+    Value elem = String::emptyString;
+    if (!O->asTypedArrayObject()->buffer()->isDetachedBuffer()) {
+        elem = O->getIndexedProperty(state, Value(0)).value(state, O);
+    }
```
Comment (clover2123) [inline src/runtime/GlobalObjectBuiltinTypedArray.cpp:1270]: Ditto. It seems that this part does not match to the spec.

File: src/runtime/GlobalObjectBuiltinTypedArray.cpp [hunk 4]
```diff
@@ -1276,8 +1278,9 @@ static Value builtinTypedArrayJoin(ExecutionState& state, Value thisValue, size_
             }
             builder.appendString(sep);
         }
-        elem = O->getIndexedProperty(state, Value(curIndex)).value(state, O);
-        RELEASE_ASSERT(!elem.isUndefinedOrNull());
+        if (!O->asTypedArrayObject()->buffer()->isDetachedBuffer()) {
+            elem = O->getIndexedProperty(state, Value(curIndex)).value(state, O);
+        }
```
Comment (clover2123) [inline src/runtime/GlobalObjectBuiltinTypedArray.cpp:1283]: ditto.

**Review by clover2123 — CHANGES_REQUESTED** [review]

File: src/runtime/GlobalObjectBuiltinTypedArray.cpp (hunk 2 above)
Comment (kisbg) [reply src/runtime/GlobalObjectBuiltinTypedArray.cpp:572]: True, i updated to the latest standart

File: src/runtime/GlobalObjectBuiltinTypedArray.cpp (hunk 3 above)
Comment (kisbg) [reply src/runtime/GlobalObjectBuiltinTypedArray.cpp:1270]: True, i updated to the latest standart

File: src/runtime/GlobalObjectBuiltinTypedArray.cpp (hunk 4 above)
Comment (kisbg) [reply src/runtime/GlobalObjectBuiltinTypedArray.cpp:1283]: ditto.

File: src/interpreter/ByteCodeInterpreter.cpp [hunk 5]
```diff
@@ -32,6 +32,7 @@
 #include "runtime/EnumerateObject.h"
 #include "runtime/ErrorObject.h"
 #include "runtime/ArrayObject.h"
+#include "runtime/TypedArrayObject.h"
```
Comment (clover2123) [inline src/interpreter/ByteCodeInterpreter.cpp:35]: Please remove this header file too.
Other seems good to me :)

Comment (clover2123) [reply src/interpreter/ByteCodeInterpreter.cpp:35]: Would you check this once again? Thanks in advance.

**Review by clover2123 — APPROVED** [review]
LGTM
